content/en/docs/guides/flux-e2e.md

```diff
@@ -0,0 +1,148 @@
+
+## Terminology
+
+Flux uses the following terms throughout the codebase and documentation:
+
+* **Cluster** - Any number of Kubernetes nodes, joined into a group to run containerized applications.
+* **Commit** - A snapshot of a Git repository's state (or any Version Control System) at any given time.
+* **Client** - Any application or resource manager which implements the "customer" side of any API or Service.
+* **Resource** - In Kubernetes, a YAML data structure represents cluster objects that drive workloads like: `Deployment`, `Pod`, `Service`, `Ingress`, `StatefulSet`, `Job`, and many others.
+* **Custom Resource** - Kubernetes provides a Custom Resource Definition (CRD) type for defining Custom Resources to be implemented by a controller. In Flux, examples include: `GitRepository`, `Bucket`, `HelmRepository`, `Kustomization`, `HelmRelease`, `Alert`, and others.
+* **Field** - YAML resources are collections of data fields, which can be nested to create complex structures like with Array and Map.
+* **Event** - A YAML resource emits events while undergoing state transitions, which themselves (`Event`) are also resources.
+* **API** - In Kubernetes, an API consists of (usually) a CRD, a control loop, and optionally one or more admission or mutation hooks. Flux's APIs are also known, collectively, as the GitOps Toolkit.
+* **Agent** - A process which runs in the cluster and does some work on behalf of users. Flux's controllers are "software agents" that implement a control loop.
+* **Service** - When Kubernetes `Deployments` spawn workloads, they are placed in ephemeral `Pods` which usually are not directly addressible. `Services` are used to connect these `Endpoints` to a stable address that can usually be discovered through a DNS lookup in the cluster.
+
+
+## Microservice architecture
+
+Flux is composed of four separable core components or controllers: [Source Controller][], [Kustomize Controller][], [Helm Controller][], and [Notification Controller][], with two extra components: [Image Automation Controller][Image reflector and automation controllers] and [Image Reflector Controller][Image reflector and automation controllers]. These controllers or Agents run on the Cluster, and they define APIs which are based on Custom Resources that altogether implement the GitOps Toolkit.
+
+**Source Controller** is the Agent responsible for pulling Commit data into the Cluster. Commits are made available as a read-only Service to Clients, which can connect with the Source Controller and fetch Artifacts, `.tar.gz` files containing Kubernetes Resource manifest data.
+
```

Comment:
Given that we also support `Bucket`, it may be worth using a more generic word instead of commit.

Reply:
I wanted to use the word Commit here because it's in the glossary above.

A goal of this document is to keep as a good entrypoint for learning about Flux without expanding every detail of Flux to every section. I think it's OK to say "commit data" here and avoid mixing up metaphors too early. We can bring in the more general word when we introduce the bucket for source controller. 👍

Thanks @pjbgf !

(But there should be a section that mentions the source controller and buckets.)

Reply:
There is now a reference to the `Bucket` kind under (4), (6). And a link out to the `BucketSpec` mentioned adjacent to where the `GitRepository` step is covered in detail under (7).